### PR TITLE
✨ Set install namespace of addonTemplate from config

### PR DESCRIPTION
**Pull request by qiujian16**
## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by qiujian16** [comment]
/hold

still need e2e test.

File: pkg/addon/templateagent/values.go [hunk 1]
```diff
@@ -87,7 +96,7 @@ func (a *CRDTemplateAgentAddon) getValues(
 	if err != nil {
 		return presetValues, overrideValues, privateValues, nil
 	}
-	overrideValues = addonfactory.MergeValues(overrideValues, builtinValues)
+	overrideValues = addonfactory.MergeValues(builtinValues, overrideValues)
```
Comment (zhujian7) [inline pkg/addon/templateagent/values.go:99]: the builtin Values are immutable

Comment (qiujian16) [reply pkg/addon/templateagent/values.go:99]: I thought we want overridevalue to override builtinValue?

Comment (zhujian7) [reply pkg/addon/templateagent/values.go:99]: we want the overridevalue to override defaultValue, but for bultinValues(ClusterName), they should be immutable.

File: pkg/addon/templateagent/values.go [hunk 2]
```diff
@@ -36,6 +36,15 @@ func ToAddOnRegistriesPrivateValues(config addonapiv1alpha1.AddOnDeploymentConfi
 	}, nil
 }
 
+func ToAddOnInstallNamespacePrivateValues(config addonapiv1alpha1.AddOnDeploymentConfig) (addonfactory.Values, error) {
+	if len(config.Spec.AgentInstallNamespace) == 0 {
+		return nil, nil
+	}
+	return addonfactory.Values{
+		"INSTALL_NAMESPACE": config.Spec.AgentInstallNamespace,
```
Comment (zhujian7) [inline pkg/addon/templateagent/values.go:44]: Not sure if the install namespace should be a private value, if it is private, it should start with `__`, like`__INSTALL_NAMESPACE`

Comment (qiujian16) [reply pkg/addon/templateagent/values.go]: this is done

File: pkg/addon/templateagent/values.go (hunk 1 above)
Comment (qiujian16) [reply pkg/addon/templateagent/values.go:100]: I am a bit confusing, I thought this func is return overrideValues that is using builtingValues as base and override by overrideValue if the key exists. The buitInvalue is immutable as input.

File: pkg/addon/templateagent/values.go [hunk 3]
```diff
@@ -60,8 +69,9 @@ func (a *CRDTemplateAgentAddon) getValues(
 	overrideValues = addonfactory.MergeValues(overrideValues, defaultValues)
```
Comment (zhujian7) [inline pkg/addon/templateagent/values.go:69]:
```suggestion
	overrideValues = addonfactory.MergeValues(defaultValues, overrideValues)
```

File: pkg/addon/templateagent/values_test.go [hunk 4]
```diff
@@ -100,3 +104,155 @@ func TestGetAddOnRegistriesPrivateValuesFromClusterAnnotation(t *testing.T) {
...
+			expectedOverride: map[string]interface{}{
+				"CLUSTER_NAME": "test-cluster",
+				"key1":         "value1",
+			},
+			expectedPrivate: map[string]interface{}{
+				InstallNamespacePrivateValueKey: "default-ns",
+			},
+		},
+		{
+			name:         "default should not be overridden",
```
Comment (zhujian7) [inline pkg/addon/templateagent/values_test.go:183]: default values can not be overridden, there are some descriptions in the tests that are not precise, I will submit another PR to refactor it after this is merged.

**Comment by zhujian7** [comment]
@qiujian16 I suggest that we add a `:warning:` to the PR title: `✨ ⚠️ Set install namespace of addonTemplate from config`, and add some description to the PR summary to state the `INSTALL_NAMESPACE` is removed, and user can use downward API, etc.

**Pull request merged**